clients/identity/authentication.go

```diff
@@ -55,7 +60,7 @@ func (api IdentityClient) CheckRequest(req *http.Request) (context.Context, int,
 
 	// if neither user nor service request, return unchanged ctx
 	if !isUserReq && !isServiceReq {
-		return ctx, http.StatusUnauthorized, nil
+		return ctx, http.StatusUnauthorized, errors.WithMessage(errUnableToIdentifyRequest, "no headers set on request")
```

Comment:
I'm a little concerned that "not providing auth headers" is considered an error.
I'm happy with `StatusUnauthorized` being returned, but why an error _too_ - shouldn't errors be reserved for failures, not for negative checks?